### PR TITLE
Activate and Deactivate App mutations

### PR DESCRIPTION
**Pull request by korycins**
- AppActivate mutation
- AppDeactivate mutation
- tests

### CODE REVIEW

File: tests/api/app/mutations/test_app_activate.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    variables = {
+        "id": id,
+    }
+    response = staff_api_client.post_graphql(
+        query, variables=variables, permissions=(permission_manage_orders,)
+    )
+    assert_no_permission(response)
+
+    app.refresh_from_db()
+    assert not app.is_active
```
Comment (IKarbowiak) [inline tests/api/app/mutations/test_app_activate.py:98]: This test is not needed, `post_graphql` is doing this check.

File: tests/api/app/mutations/test_app_deactivate.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    variables = {
+        "id": id,
+    }
+    response = staff_api_client.post_graphql(
+        query, variables=variables, permissions=(permission_manage_orders,)
+    )
+    assert_no_permission(response)
+
+    app.refresh_from_db()
+    assert app.is_active
```
Comment (IKarbowiak) [inline tests/api/app/mutations/test_app_deactivate.py:98]: Same here, `post_graphql` is doing this check.

File: saleor/graphql/app/mutations.py [hunk 3]
```diff
@@ -208,6 +211,75 @@ def clean_instance(cls, info, instance):
...
+            msg = "App is already deactivated."
+            code = AppErrorCode.INVALID.value
+            raise ValidationError({"id": ValidationError(msg, code=code)})
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        app = cls.get_instance(info, **data)
+        cls.clean_instance(info, app)
+        app.is_active = False
+        cls.save(info, app, cleaned_input=None)
```
Comment (fowczarek) [inline saleor/graphql/app/mutations.py:279]: Maby? 
```suggestion
        app.save(update_fields=["is_active"])
```

File: tests/api/app/mutations/test_app_activate.py (hunk 1 above)
Comment (fowczarek) [reply tests/api/app/mutations/test_app_activate.py:98]: We have tasks to remove this "feature" from post_graphql.

File: saleor/graphql/app/mutations.py [hunk 4]
```diff
@@ -208,6 +211,75 @@ def clean_instance(cls, info, instance):
...
+            permissions
+        ):
+            msg = "You don't have enough permission to perform this action."
+            code = AppErrorCode.OUT_OF_SCOPE_APP.value
+            raise ValidationError({"id": ValidationError(msg, code=code)})
+
+        if app.is_active:
+            msg = "App is already active."
+            code = AppErrorCode.INVALID.value
+            raise ValidationError({"id": ValidationError(msg, code=code)})
```
Comment (maarcingebala) [inline saleor/graphql/app/mutations.py:239]: I'm wondering, is it useful to raise this validation error? What if the mutation just did nothing and returned the unmodified object?

File: saleor/graphql/app/mutations.py [hunk 5]
```diff
@@ -208,6 +211,75 @@ def clean_instance(cls, info, instance):
...
+    @classmethod
+    def clean_instance(cls, info, app: App):
+        requestor = get_user_or_app_from_context(info.context)
+        permissions = app.permissions.all()
+        if not requestor_is_superuser(requestor) and not requestor.has_perms(
+            permissions
+        ):
+            msg = "You don't have enough permission to perform this action."
+            code = AppErrorCode.OUT_OF_SCOPE_APP.value
+            raise ValidationError({"id": ValidationError(msg, code=code)})
```
Comment (maarcingebala) [inline saleor/graphql/app/mutations.py:234]: I'm a bit lost on why we prefer `ValidationError` and not `PermissionDenied` here.

File: saleor/graphql/app/mutations.py (hunk 3 above)
Comment (korycins) [reply saleor/graphql/app/mutations.py:279]: Not sure of this. ModelMutation has its own flow, if we are able then we should use it but you are right - having `update_fields` in `cls.save()` would be a good idea.

File: saleor/graphql/app/mutations.py (hunk 5 above)
Comment (korycins) [reply saleor/graphql/app/mutations.py:234]: @maarcingebala I just aligned to the current flow here.
PermissionDenied is dedicated to raising when the user/app doesn't have the required permission for a given section. Here, we use ValidationError as this error is not connected only with the user. The user has access to Apps, but he can't manage Apps with bigger permissions than he has for his own.  
If you don't like it then what do you propose?

**Pull request merged**